src/Sparrow.Server/Platform/Posix/CGroupHelper.cs

```diff
@@ -0,0 +1,315 @@
+    private static string FindCGroupPathForSubsystem(Predicate<IEnumerable<string>> isSubSystem)
+    {
+        foreach (var line in File.ReadLines(PROC_CGROUP_FILENAME))
+        {
+            var match = FindCGroupPathForSubsystemReg.Match(line);
+            if(match.Success == false)
+                throw new CGroupException($"Failed to parse cgroup info file contents - {line}.");
+                
+            if(isSubSystem(match.Groups["subsystem_list"].Captures.Select(x => x.Value)) == false)
+                continue;
+
+            return match.Groups["path"].Value;
+        }
+
+        return null;
+    }
+}
+
+public class CGroupV2 : CGroup
+{
+    protected override string MemoryLimitFileName => "memory.max";
+    protected override string MemoryUsageFileName => "memory.current";
+    protected override string MaxMemoryUsageFileName => "memory.peak";
+
+    protected override string FindCGroupPathForMemory() => FindCGroupPath();
+    
+    private static string FindCGroupPath()
+    {
+        FindHierarchyMount(out var mountRoot, out var mountPath);
+        var pathForSubsystem = FindCGroupPathForSubsystem();
+        
+        return CombinePaths(mountRoot, mountPath, pathForSubsystem);
+    }
+    private static void FindHierarchyMount(out string mountRoot, out string mountPath)
+    {
+        foreach (var match in FindHierarchyMount())
+        {
+            mountRoot = match.Groups["mountroot"].Value;
+            mountPath = match.Groups["mountpath"].Value;
+            return;
+        }
+
+        throw new CGroupException($"Couldn't find hierarchy mount in {PROC_MOUNTINFO_FILENAME}");
+    }
+    
+    // 0::/user.slice/user-1000.slice/user@1000.service/apps.slice/apps-org.gnome.Terminal.slice/vte-spawn-d7794050-ce4a-451b-92c2-a2433019409e.scope
+    private static readonly Regex FindCGroupPathForSubsystemReg = new Regex(@"^\d+::(?<path>.*)$", RegexOptions.Compiled);
+    private static string FindCGroupPathForSubsystem()
+    {
+        foreach (var line in File.ReadLines(PROC_CGROUP_FILENAME))
+        {
+            var match = FindCGroupPathForSubsystemReg.Match(line);
+            if(match.Success == false)
+                continue;
+                
+            return match.Groups["path"].Value;
+        }
+
+        return null;
+    }
+}
+
+public class UnidentifiedCGroup : CGroup
+{
+    private readonly string _errorMessage;
+    private DateTime _lastLog = DateTime.MinValue;
+    protected override string MemoryLimitFileName => null;
+    protected override string MemoryUsageFileName => null;
+    protected override string MaxMemoryUsageFileName => null;
+
+    public UnidentifiedCGroup(string errorMessage)
+    {
+        _errorMessage = errorMessage;
+    }
+
+    protected override string FindCGroupPathForMemory()
+    {
+        if (_lastLog + TimeSpan.FromMinutes(10) < DateTime.UtcNow)
```

Comment:
I think we have `SystemTime` class for UtcNow? Please consider changing DateTime occurences to that.

Reply:
I see it belongs to `Raven.Client` and the `Sparrow.Server` doesn't have a reference to it.
Moving that will add a lot of mess to the PR and I don't see much benefit so I think it doesn't worth it.

Reply:
Understood. That ok with you @ppekrol ?

Reply:
Fine